lib/devise/models/database_authenticatable.rb

```diff
@@ -37,17 +37,18 @@ def self.required_fields(klass)
       # the hashed password.
       def password=(new_password)
         @password = new_password
-        self.encrypted_password = password_digest(@password) if @password.present?
+        self.encrypted_password = password_digest(@password) 
       end
 
       # Verifies whether a password (ie from sign in) is the user password.
       def valid_password?(password)
+        return false if password.blank?
```

Comment:
I believe we can remove this condition since it's already present [inside](https://github.com/plataformatec/devise/blob/master/lib/devise/encryptor.rb#L15) `Devise::Encryptor.compare`

Reply:
No, `Devise::Encryptor.compare` is checking that `encrypted_password` is present (or at least it is [as of now](https://github.com/plataformatec/devise/blob/4656e96d9ebd3947ceee27e698aee76393dc578f/lib/devise/encryptor.rb#L15)).

Reply:
Yeah, but since we're returning `nil` from `#password_digest` when the `password` is blank, `encrypted_password` will always be `nil`, and the `.compare` method will return in the ` return false if hashed_password.blank?` condition.